pkg/cmd/roachtest/roachtestutil/mixedversion/steps.go

```diff
@@ -205,7 +205,7 @@ func (s restartWithNewBinaryStep) Run(
 		l,
 		h.runner.cluster,
 		h.runner.cluster.Node(s.node),
-		startOpts(option.WithInitTarget(s.initTarget)),
+		startOpts(option.WithInitTarget(s.initTarget), option.WaitForReplication()),
```

Comment:
Note the number of ranges can get quite large especially during longer runs, involving data-generating workloads. Also, the immediate node being restarted is now force to wait despite the fact that none of its ranges may be under-replicated.

Reply:
I put a 30 minute timeout on this wait. In addition, tests are now able to disable this behaviour if they load a lot of data or if they inject some kind of failure that leads to expected replication issues.